Task 2/main.py

```diff
@@ -0,0 +1,37 @@
+# Turimas "users" masyvas. 
+
+# Parašykite funkcijas, kurios atlikas nurodytas užduotis:
+# 1. funkcija "getUserMedianAge" - kaip argumentą priims masyvą ir duoto masyvo 
+# atveju grąžins visų "users" amžiaus medianą.
+# 2. funkcija "getOldestUser" -  kaip argumentą priims masyvą ir duoto masyvo 
+# atveju grąžins vyriausio vartotojo vardą.
+
+users = [
+  { "id": '1', "name": 'John Smith', "age": 20 },
+  { "id": '2', "name": 'Ann Smith', "age": 24 },
+  { "id": '3', "name": 'Tom Jones', "age": 31 },
+  { "id": '4', "name": 'Rose Peterson', "age": 17 },
+  { "id": '5', "name": 'Alex John', "age": 25 },
+  { "id": '6', "name": 'Ronald Jones', "age": 63 },
+  { "id": '7', "name": 'Elton Smith', "age": 16 },
+  { "id": '8', "name": 'Simon Peterson', "age": 30 },
+  { "id": '9', "name": 'Daniel Cane', "age": 51 },
+]
+
+def median (numbers):
+  length = len(numbers)
+  index = length // 2
+  if length % 2:
+    return sorted(numbers)[index]
+  return sum(sorted(numbers)[index - 1:index + 1]) / 2
+
+def getUserMedianAge(users):
+  return median([user["age"] for user in users])
+
+def getOldestUser(users):
+  return sorted(users, key=lambda x: x["age"])[-1]["name"]
```

Comment:
Reikėtų tiksliai pamatuoti, bet didžiausią narį turėtų būti efektyviau rasti nei visą sąrašą eilės tvarka sudėlioti.
```suggestion
  oldestUser = max(users, key=lambda user: user['age'])
  return(oldestUser['name'])
```